### PR TITLE
#162727483 Implement like specific comment

### PR DESCRIPTION
**Pull request by abejide001**
#### What does this PR do?
Enable user to like and unlike a specific comment
#### Description of Task to be completed?
- Implement like method in comment reaction controller
- Implement unlike method in comment reaction controller
- Create comment reaction summary 
- Implement Number of likes logic for a specific comment
#### How should this be manually tested?
- Run `npm i` to install application dependencies
- Visit the endpoint `localhost:9000/api/v1/comments/2/like` to like a specific comment
- Visit the endpoint `localhost:9000/api/v1/comments/2/unlike` to unlike a specific comment
#### Any background context you want to provide?
- The endpoint is only accessible to authenticated users
- A post can only be liked once
#### What are the relevant pivotal tracker stories?
[#162727483](https://www.pivotaltracker.com/story/show/162727483)
#### Screenshots (if appropriate)
screen shot 2019-01-24 at 3 49 31 pm
screen shot 2019-01-24 at 3 49 48 pm
screen shot 2019-01-24 at 12 59 21 pm

### CODE REVIEW

**Review by julietezekwe — APPROVED** [review]
ok

**Review by daniellamarr — APPROVED** [review]

File: src/db/service/commentReactionSummary.js [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+    });
+    const findLikeSummary = await CommentReactionSummary.findOne({
+      where: {
+        commentId: id,
+      }
+    });
+    if (findLikeSummary) {
+      const likes = findLikeSummary.noOfLikes + 1;
+      findLikeSummary.noOfLikes = likes;
+      findLikeSummary.save();
```
Comment (andela-dbamidele) [inline src/db/service/commentReactionSummary.js:30]: Are you sure this works?

Comment (abejide001) [reply src/db/service/commentReactionSummary.js:30]: yes, the implementation works.

File: src/controllers/CommentReactionController.js [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+  /**
+   * @static
+   * @param {Object} req
+   * @param {object} res
+   * @param {object} done
+   * @return {object} comment
+   */
+  static async likeComment(req, res, done) {
+    try {
+      const comment = await Comment.findById(req.params.commentId);
```
Comment (andela-dbamidele) [inline src/controllers/CommentReactionController.js]: Please put this in the middleware

File: src/controllers/CommentReactionController.js [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+          'Not Found',
+          404,
+          'This comment does not exist',
+        );
+        return res.status(response.code).json(response);
+      }
+      LikeUnlikeComment.like(req.params.commentId);
+      CommentReaction.findOrCreate({
+        where: {
+          $and: [{ commentId: req.params.commentId }, { userId: req.verifyUser.id }]
```
Comment (andela-dbamidele) [inline src/controllers/CommentReactionController.js]: you can simplify this by doing

```
where: {
  commentId: ....,
  userId: ....,
}
```

File: src/controllers/CommentReactionController.js [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        },
+        defaults: {
```
Comment (andela-dbamidele) [inline src/controllers/CommentReactionController.js:23]: Why do you need this? Looks like repetition to me.

**Review by andela-dbamidele — CHANGES_REQUESTED** [review]

File: src/controllers/CommentReactionController.js (hunk 2 above)
Comment (abejide001) [reply src/controllers/CommentReactionController.js]: okay, fixed.

File: src/controllers/CommentReactionController.js (hunk 3 above)
Comment (abejide001) [reply src/controllers/CommentReactionController.js]: okay, thanks.

File: src/controllers/CommentReactionController.js (hunk 4 above)
Comment (abejide001) [reply src/controllers/CommentReactionController.js:23]: the defaults? I am passing in the userId and commentId into the database, I don't think there is any repetition there

File: src/routes/index.js [hunk 5]
```diff
@@ -19,6 +20,7 @@ router.use('/rate', TokenAuthenticate.tokenVerify, ratingRouter);
 router.use('/api-docs', swaggerUi.serve, swaggerUi.setup(swaggerSpec));
 router.use('/arts', artsRoute);
 router.use('/arts/comments/', commentRouter);
+router.use('/comments', commentReaction);
```
Comment (wombolo) [inline src/routes/index.js]: I think this file will be cleaner if you make use of the existing comments route.

File: src/controllers/CommentReactionController.js (hunk 4 above)
Comment (andela-dbamidele) [reply src/controllers/CommentReactionController.js:23]: What is the default doing exactly?

Comment (abejide001) [reply src/controllers/CommentReactionController.js:23]: I'm specifying the default values to be inserted into the database fields that's the `userId` and the `commentId`, those are the values that will be inserted whether it is updating or creating

**Review by wombolo — APPROVED** [review]
LGTM

File: src/routes/index.js [hunk 6]
```diff
@@ -22,6 +23,7 @@ router.use('/api-docs', swaggerUi.serve, swaggerUi.setup(swaggerSpec));
 router.use('/arts', artsRoute);
 router.use('/artsreport', reportRouter);
 router.use('/arts/comments/', commentRouter);
+router.use('/comments', commentReaction);
```
Comment (andela-dbamidele) [inline src/routes/index.js]: How's this different from the one above it?

Comment (abejide001) [reply src/routes/index.js]: the routes are different, `/comments/:id/like` to like a specific comment, and `/arts/comments` to comment on an art

**Pull request merged**